extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java

```diff
@@ -281,6 +283,16 @@ private void checkProperConfigAccessInProvider(AnnotationInstance instance) {
         }
     }
 
+    private void checkProperConstructorInProvider(AnnotationInstance i) {
+        ClassInfo targetClass = i.target().asClass();
+        if (!targetClass.hasNoArgsConstructor()
+                || targetClass.methods().stream().filter(m -> m.name().equals("<init>")).count() > 1) {
```

Comment:
I wonder if we should send a PR to jandex to add a method like `ClassInfo#constructors()` ;-).

Reply:
That's an excellent idea! I'll do that later today :)

Reply:
Here it is: https://github.com/wildfly/jandex/pull/78